src/lib.rs

```diff
@@ -283,12 +311,12 @@ pub fn eval_beat<T: Into<Val>>(program: &Program, t: T) -> Result<Val, ()> {
             Fg(..) | Bg(..) | Khz(..) | Comment(..) => (),
         }
     }
-    stack.pop().ok_or(())
+    stack.pop().unwrap()
 }
 
-pub fn parse_beat(text: &str) -> Result<Vec<Cmd>, &str> {
+pub fn parse_beat(text: &str) -> Result<Code, &str> {
```

Comment:
I think we should be able to parse semantically invalid code, for instance we could parse suffixes that need to be attached to an existing program.